core/src/wallet/ethereum/EthereumLikeWallet.cpp

```diff
@@ -148,6 +148,22 @@ namespace ledger {
             });
         }
 
+        static DerivationScheme getAccountScheme(DerivationScheme &scheme) {
+            auto accountScheme = scheme.getSchemeTo(DerivationSchemeLevel::ACCOUNT_INDEX);
+            // To handle all exotic paths we should avoid private derivations
+            // So if node or/and address level are hardened, then they are included in account's derivation path
+            auto path = scheme.getPath();
+            auto hardenedDepth = path.getDepth();
+            while (hardenedDepth) {
+                if (path.isHardened(hardenedDepth - 1)) {
+                    break;
+                }
+                hardenedDepth --;
```

Comment:
Can you use either `hardenedDepth--` or `--hardenedDepth` but not `hardenedDepth --`, please? 🙂